phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -200,12 +224,12 @@ public Expression visitLeave(ComparisonParseNode node, List<Expression> children
         Expression lhsExpr = children.get(0);
         Expression rhsExpr = children.get(1);
         PDataType dataTypeOfLHSExpr = lhsExpr.getDataType();
-        if (dataTypeOfLHSExpr != null && !dataTypeOfLHSExpr.isEqualitySupported()) {
+        if (!(lhsExpr instanceof BaseJSONExpression)&& dataTypeOfLHSExpr != null && !dataTypeOfLHSExpr.isEqualitySupported()) {
```

Comment:
Hello,James,because some JSON operators will return data of other data type(like VARCHAR,INTEGER)

for example : 
the following operation : 
```
'{"a":3 , "b":"string"}' -> 'a'
```
this type will be as integer
but another operation :
```
'{"a":"string","b":2.3}' -> 'a'
```
this type will be as varchar

but we don't know real data type here,so I thought we need skip checking those operators here.